src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1739,6 +1748,47 @@ bool OutputStateMachineEngine::_GetRepeatCount(std::basic_string_view<size_t> pa
     return success;
 }
 
+// Routine Description:
+// - Parse OscSetClipboard parameters with the format `Pc;Pd`. Currently the first parameter `Pc` is
+// ignored. The second parameter `Pd` should be a valid base64 string or character `?`.
+// Arguments:
+// - string - Osc String input.
+// - content - Content to set to clipboard.
+// - queryClipboard - Whether to get clipboard content and return it to terminal with base64 encoded.
+// Return Value:
+// - True if there was a valid base64 string or the passed parameter was `?`.
+bool OutputStateMachineEngine::_GetOscSetClipboard(const std::wstring_view string,
+                                                   std::wstring& content,
+                                                   bool& queryClipboard) const noexcept
+{
+    bool success = false;
+    size_t pos = string.find(';');
+    if (pos != std::wstring_view::npos)
+    {
+        std::wstring_view substr = string.substr(pos + 1);
+        if (substr == L"?")
+        {
+            queryClipboard = true;
+            success = true;
+        }
+        else
+        {
+            try
+            {
+                auto buffer = CryptographicBuffer::DecodeFromBase64String(winrt::hstring(substr));
+                auto reader = DataReader::FromBuffer(buffer);
+                content = reader.ReadString(buffer.Length());
+                success = true;
+            }
+            catch (...)

```

Comment:
@zadjii-msft  I wanted to use CATCH_LOG here, but when an invalid base64 comes, it will freeze the terminal and make it unresponsive. I don't know the exact reason, maybe WIL cannot parse exceptions from `winrt::Windows::Security::Cryptography` well?

Reply:
Huh, that's definitely weird...

I bet it's because the winrt header that defines `winrt::Windows::Security::Cryptography` is getting included _after_ WIL - IIRC, the order matters for these things. Maybe if you move the header to before `wil` in `precomp`, it might work?

I'm totally spitballing here

Reply:
Tried with no luck. :(

Reply:
I figured this out! If you include `<wil/cppwinrt.h>` before the other `winrt` headers, you can use `CATCH_LOG` here! There's a comment with more details in the `precomp.h` file.